apis/reaper/v1alpha1/reaper_types.go

```diff
@@ -81,6 +82,34 @@ type ReaperDatacenterTemplate struct {
 	InitContainerSecurityContext *corev1.SecurityContext `json:"initContainerSecurityContext,omitempty"`
 }
 
+type ContainerImage struct {
```

Comment:
I'm not creating a common type for both Reaper and Stargate because even if the struct is similar, the kubebuilder tags are not. If anyone has a better idea let me know.

Reply:
Did you look at how images are handled now in cass-operator? Defaults are supplied via a ConfigMap. One nice thing about this is that it doesn't require any code changes if the user wants to change those defaults. We would want to do #63 for the ConfigMap approach (if we were to be consistent with the way it's done in cass-operator).

Reply:
@jsanda I did, and frankly it looks overly complex. What exactly would you like to see implemented in k8ssandra-operator? I see a few problems trying to port cass-operator's image handling code here:

- Most types are too specific to cass-operator, e.g. [this one](https://github.com/k8ssandra/cass-operator/blob/65e47b5ac16adfccfbb335abf9723263554fc3ee/apis/config/v1beta1/imageconfig_types.go#L46-L56).
- I don't see ConfigMaps being used, however I do see a field named `ImageConfigFile` but I am struggling to understand how it works since there is no test at all for it.
- Indeed we would need to do #63 before going further in this direction. Should I close this PR for now and come back to it later?

Reply:
The ConfigMap gets generated. See [here](https://github.com/k8ssandra/cass-operator/blob/master/config/manager/kustomization.yaml#L7). We can revisit using a ImageConfig/ConfigMap when we tackle #63. I will do another review of this today so we can get it merged.

Reply:
I wonder if the `+kubebuilder:default` annotation could help avoid duplicating `ContainerImage`. It says that it accepts an object as one of the types for a value. Might be worth exploring.

Reply:
As agreed offline, I'm going to merge now and investigate this idea in another PR.